### PR TITLE
Fix #245 : hitsPerPage not parsed as a number

### LINKED ISSUE #245: HitsPerPage attribute not parsed as a number
Linked to https://github.com/algolia/instantsearch.js/issues/366

### PR DESCRIPTION
**Pull request by bobylito**
This PR also contains : 
  - adds missing attribute minProximity in SearchParameters
  - updates links (were outdated)
  - adds other missing parameters
  - adds missing documentation for filter on SearchParameters

### CODE REVIEW

File: src/SearchParameters/index.js [hunk 1]
```diff
@@ -358,8 +363,8 @@ function SearchParameters(newParameters) {
     if (!this.hasOwnProperty(paramName)) {
       // IE8/9 has no console (BUT if devtools opened), nevermind there's no
       // developer working ONLY in IE8/9
-      window.console &&
-      window.console.error(
+      console &&
```
Comment (redox) [inline src/SearchParameters/index.js]: why replacing `window.console` by `console`? That code won't work on old IE.

Comment (bobylito) [inline src/SearchParameters/index.js]: Because it rely on window which breaks which unit testing with node. I will improve that.

**Comment by bobylito** [comment]
Tests are still broken. Weird thing is that it's on MS Edge about 400 answer from the server...

**Comment by vvo** [comment]
If tests are I need to work a little on the test stack indeed. Uf ecerything seems fine and tests ok in node chrome and change is a simple one, merge

**Comment by bobylito** [comment]
I wouldn't go as far as to say it is trivial but it's pretty simple so I guess it's good to merge :) (and the failing test is already in previously merged and released work :P )

**Comment by bobylito** [comment]
Fixes #245

**Comment by vvo** [comment]
Ah yes that's just IE8 failing, the helper is no more compatible. Once we release instantsearch I will need to work a lil bit on this.

**Comment by vvo** [comment]
Can you think of any application relying on hitsPerPage being a string that we would break?

**Comment by bobylito** [comment]
https://www.algolia.com/doc/rest#param-hitsPerPage it is an integer. So it will break one way or another, if it is not parseable

**Comment by vvo** [comment]
LGTM

**Pull request merged**